### PR TITLE
Sign In with Apple

### PR DESCRIPTION
**Pull request by lucjross**
https://github.com/markbates/goth/issues/290

### CODE REVIEW

**Comment by bentranter** [comment]
Thanks @lucjross! Pretty sure the test failure wasn't your fault, so I've restarted the job that failed. I'll merge once that passes.

File: providers/apple/apple.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+func (Provider) UnmarshalSession(data string) (goth.Session, error) {
+	s := &Session{}
+	err := json.NewDecoder(strings.NewReader(data)).Decode(s)
+	return s, err
+}
+
+func (Provider) FetchUser(goth.Session) (goth.User, error) {
+	return goth.User{}, errors.New("not implemented")
```
Comment (lucjross) [inline providers/apple/apple.go]: I couldn't find a user profile endpoint for Apple.

**Comment by lucjross** [comment]
Please, don't merge yet. We want to try and get FetchUser implemented.

**Comment by bentranter** [comment]
Sure, no problem @lucjross, I'll wait.

**Comment by lucjross** [comment]
@bentranter, I still haven't found a user profile endpoint for Apple, but I made FetchUser a little better by using the unique ID from the ID token on the access token response.

Looks like you're trying to support Go versions down to 1.7. If so I'll need to pull out that `jwx` library.

**Comment by bentranter** [comment]
@lucjross I wouldn't worry about supporting all the way back to 1.7, since they're all way past EOL – I probably should've dropped support for them a long time ago. If you want, you can remove the failing old versions from the test matrix, if not, I can do it and you can rebase your PR.

**Comment by lucjross** [comment]
Cool, thanks for the flexibility on that. I think we've implemented everything needed on our end.

**Pull request merged**